### PR TITLE
Redesigned About dialog

### PR DESCRIPTION
**Pull request by in3otd**
~~**Do not merge yet, this is still a WIP**~~
**Edit: cleaned up a bit the commits, should be ok now.**

Here is the redesigned About dialog, as per #172 . Please take a look at the dialog and code and check if it's ok. Might still need some (minor) fixes.

qucsabout

### CODE REVIEW

File: qucs/qucs/dialogs/aboutdialog.cpp [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  QLabel *lbl;
+
+  setWindowTitle(tr("About Qucs"));
+  
+  all = new QVBoxLayout(this);
+  //all->setContentsMargins(0,0,0,0);
+  //all->setSpacing(0);
+
+  QLabel *iconLabel = new QLabel();
+  iconLabel->setPixmap(QPixmap(QString("qucs/bitmaps/hicolor/128x128/apps/qucs.png")));
```
Comment (nvdl) [inline qucs/qucs/dialogs/aboutdialog.cpp]: Possible this will be a resource reference later.

**Comment by nvdl** [comment]
I was polishing the logo for the new dialog.
Possibly, you can add to it.

logo

File: qucs/qucs/dialogs/aboutdialog.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+   Q_OBJECT
+
+public:
+  AboutDialog(QWidget *parent = 0);
+
+private:
+  void setAuthorsText(void);
+  void setTrText(void);
+
+  std::array<QString, 8> currAuths; // current Qucs authors
```
Comment (guitorri) [inline qucs/qucs/dialogs/aboutdialog.h]: `std::array` seems not yet available in clang 3.4... 
https://travis-ci.org/Qucs/qucs/jobs/61680030#L7991

**Comment by guitorri** [comment]
We should unify the artwork on logos, icons and website (pick fonts and colors)... I scrapped some files from the mailing list I will put them somewhere accessible. However the ones I have are not SVG files... @nvdl which font did you use?

**Comment by nvdl** [comment]
_polishing the logo_

It is already on the web, just added some gradient to give a 3D feel.
No idea about the font though. But if we agree on fonts, color schemes, etc. I can make new logos.

What is this stuff going on? Any Wiki (or something) where we can propose?

File: qucs/qucs/dialogs/aboutdialog.h (hunk 2 above)
Comment (in3otd) [inline qucs/qucs/dialogs/aboutdialog.h]: oh, well... but with the clang 3.3 I'm using here it compiles fine (!). I think I forgot an `#include <array>` anyway, let me check...

**Comment by in3otd** [comment]
I have cleaned up the code a little, should be good to merge now, please check.

File: qucs/qucs/dialogs/aboutdialog.cpp [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+  authorsBrowser = new QTextBrowser;
+  trBrowser = new QTextBrowser;
+
+
+
+  QString supportText;
+  // link to home page, help mailing list, IRC ?
+  supportText = tr("Home Page") + " : <a href='http://qucs.sourceforge.net/'>http://qucs.sourceforge.net/</a><br/>"+
+    tr("Documentation start page") + " : <a href='http://qucs.sourceforge.net/docs.html'>http://qucs.sourceforge.net/docs.html</a><br/>" +
+    tr("Components reference manual") + " : <a href='http://qucs.sourceforge.net/doc/0.0.19/html/index.html'>http://qucs.sourceforge.net/doc/0.0.19/html/index.html</a><br/>" +
```
Comment (guitorri) [inline qucs/qucs/dialogs/aboutdialog.cpp]: This link is dead.
I moved the component reference to https://github.com/guitorri/qucs-manual. Perhaps we can move it to the Qucs organization. It is still needs some work thou.
Perhaps the [Readme.md resources](https://github.com/Qucs/qucs#resources) would fit here nicely.

**Comment by guitorri** [comment]
This looks great! 
Two other comments:
- @in3otd consider putting yourself on the list of authors;
- perhaps the font resize can be switched off on the QTextBrowser. I hit ctrl+scroll by accident and the text became very small...

**Comment by in3otd** [comment]
ok, which link should I use then for the Component Reference Manual? Will you put it back on SF or keep it just in http://guitorri.github.io/qucs-manual/ ? It's a WIP, as many things here, but already quite good IMHO.
Will look into the latest two comments also.

**Comment by guitorri** [comment]
I will move the qucs-manual to the Qucs organization. I just want to avoid publishing links that will soon be broken.

**Comment by guitorri** [comment]
Done, moved to https://github.com/Qucs/qucs-manual

**Comment by guitorri** [comment]
yes, the webpage: http://qucs.github.io/qucs-manual/

**Comment by in3otd** [comment]
updated the authors list, updated the doc link and the text in the Support page and disabled the Ctrl-Wheel event. Took a while to find out who was actually handling it (not the QTextBrowser but its viewport). Should be ok now.

**Comment by guitorri** [comment]
Very nice! Congrats!

**Pull request merged**